internal/extramanifest/merge.go

```diff
@@ -0,0 +1,406 @@
+// nolint
+package extramanifest
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+
+	apiRes "k8s.io/apimachinery/pkg/api/resource"
+)
+
+// Code copied from https://github.com/open-cluster-management-io/config-policy-controller/blob/main/controllers/configurationpolicy_controller.go
+// and https://github.com/open-cluster-management-io/config-policy-controller/blob/main/controllers/configurationpolicy_utils.go for
+// merging manifests the ACM way
```

Comment:
Everything in this file is cloned without modification, correct? That may also be good to note.

Actually, it may be good to update the URLs in the comment with the permalink instead, so that we have a reference to the exact version that was cloned. ie.
https://github.com/open-cluster-management-io/config-policy-controller/blob/d734b7aa1ec2ae936edb48ec622b29ac70871921/controllers/configurationpolicy_controller.go